src/llm_utils.py

```diff
@@ -5,178 +5,136 @@
 from google import genai
 from typing import List, Dict
 
+from github_utils import get_valid_hunks
+
 def analyze_code_changes(diff_content: str) -> List[Dict]:
     """
     Analyze code changes using OpenAI's GPT model
     Returns a list of review comments
     """
-    API_KEY = os.getenv('GEMINI_API_KEY')
 
-    if not API_KEY:
-        raise ValueError("GEMINI_API_KEY is not set. Please add it to your environment variables.")
+    try:
 
-    # Prepare the prompt for the LLM
-    prompt = f"""
-    Analyze the following code changes and provide review comments using EXACTLY this format:
+        API_KEY = os.getenv('GEMINI_API_KEY')
 
-    FILE: [file-path]
-    LINE: [line-number]
-    COMMENT: [your comment]
-    SUGGESTION: [optional suggested code]
+        if not API_KEY:
+            raise ValueError("GEMINI_API_KEY is not set. Please add it to your environment variables.")
 
-    ---
-    Example:
-    FILE: src/app.py
-    LINE: 42
-    COMMENT: Avoid magic numbers, consider using a constant
-    SUGGESTION: MAX_RETRIES = 3
+        hunks = get_valid_hunks(diff_content)
 
-    ---
+        # Prepare the prompt for the LLM
+        prompt = f"""
+        Analyze these code changes and provide feedback using EXACTLY this format:
 
-    Make sure to insert '---' in between two review comments. IT IS A MUST!!!
+        FILE: [file-path]
+        LINES: [start_line-end_line]  # MUST be within the same diff hunk
```

Comment:
Fetch hunks using the helper function.
```suggestion
|
```
N/A